File: Anagrams/Anagrams.cpp

class Solution {
public:
    vector<string> anagrams(vector<string> &strs) {
        map<string,int>mp;
        vector<string>ans;
        for(int i=0;i<strs.size();i++)
        {
            string temp = strs[i];
            sort(temp.begin(),temp.end());
            if(mp.find(temp) == mp.end())
                mp[temp] = i;
            else
            {
                ans.push_back(strs[i]);
                if(mp[temp]!=-1)
                {
                    ans.push_back(strs[mp[temp]]);
                    mp[temp]=-1;
                    
                }
            }
        }
        return ans;
    }
};


 

File: Anagrams/Anagrams.py


class Solution:
    # @param strs, a list of strings
    # @return a list of strings
    def anagrams(self, strs):
        dict = {}
        for word in strs:
            temp = ''.join(sorted(word))
            dict[temp] = [word] if temp not in dict else dict[temp] + [word]
        ans = []
        for word in dict:
            if len(dict[word]) >= 2:
                ans += dict[word]
        return ans
 